components/autocomplete/index.js

```diff
@@ -34,86 +99,179 @@ class Autocomplete extends Component {
 		this.bindNode = this.bindNode.bind( this );
 		this.select = this.select.bind( this );
 		this.reset = this.reset.bind( this );
-		this.onBlur = this.onBlur.bind( this );
 		this.search = this.search.bind( this );
 		this.setSelectedIndex = this.setSelectedIndex.bind( this );
 
 		this.state = this.constructor.getInitialState();
+		this.allOptions = {};
 	}
 
 	bindNode( node ) {
 		this.node = node;
 	}
 
 	select( option ) {
-		const { onSelect } = this.props;
+		const { open, range, lookup } = this.state;
+		const { onSelect } = open;
 
 		this.reset();
 
 		if ( onSelect ) {
-			onSelect( option );
+			onSelect( option.value, range, lookup );
 		}
 	}
 
 	reset() {
 		this.setState( this.constructor.getInitialState() );
 	}
 
-	onBlur( event ) {
```

Comment:
Explanation:
onBlur was removed because the aim of hiding the popup when clicking away could be better handled by using the onClickOutside function of Popover.

Reply:
Problem here is if the user tabs away from the block while the autocomplete is still open, it remains open.

Reply:
Well the original (as in the version in master) was hiding the Popover when you clicked on it meaning that options could only be selected by pressing enter. I will try to find a way to avoid both problems though.

Reply:
I ended up adding a handler for the tab key to close the menu.

Reply:
I realized that the left and right arrow keys had the same problem so they also close the menu now.

Reply:
>Well the original (as in the version in master) was hiding the Popover when you clicked on it meaning that options could only be selected by pressing enter.

Ah yeah, seeing this now on master. I can't say for certain that detecting specific keys as an implied blur is sufficient though. Do these differ at all by browser, user preferences, OS? Might be missing some like Escape.

Actually, in the case of Escape it may be working correctly, but only because [Popover stops propagation when handling this itself](https://github.com/WordPress/gutenberg/blob/ec8df21abb643cb5e67cd998269d53c0af61e654/components/popover/index.js#L198). Quite fragile.

Seems like the problem is that since the popover is rendered elsewhere in the DOM, `this.node.contains` will return `false` when clicking on a button within the popoover. But since our component here is the one responsible for rendering the list, maybe a solution is to assign a ref for the `ul` and check whether the related target is also (or instead) in this list.

Might be a bug worth addressing in a separate pull request.